plugins/modules/system/sap_task_list_execute.py

```diff
@@ -0,0 +1,437 @@
+    parameter_list = []
+    raw_task_list = []
+    task_list = []
+    if task_parameters is not None:
+        for parameters in task_parameters:
+            if parameters[0].startswith('CL_'):
+                length = len(parameters)
+                raw_task_list = raw_task_list + [{'TASKNAME': parameters[0]}]
+                task_list = [i for n, i in enumerate(raw_task_list) if i not in raw_task_list[n + 1:]]
+                if length == 3:
+                    parameter_list = parameter_list + [{'TASKNAME': parameters[0], 'FIELDNAME': parameters[1], 'VALUE': parameters[2]}]
+            else:
+                return False
+    #
+    return [{'parameter_list': parameter_list}, {'task_list': task_list}]
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['show', 'present']),
+            # values for connection
+            conn_username=dict(type='str', required=True),
+            conn_password=dict(type='str', required=True, no_log=True),
+            host=dict(type='str', required=True),
+            sysnr=dict(type='str', default="00"),
+            client=dict(type='str', default="000"),
+            # values for execution tasks
+            task_to_execute=dict(type='str', required=True),
+            task_parameters=dict(type='list', elements='list', required=False),
+            task_settings=dict(type='list', elements='str', default=['BATCH']),
+            tasks_skip=dict(type='bool', default=False),
+        ),
+        supports_check_mode=False,
+    )
+    result = dict(changed=False, msg='', results=[], out={}, stdout={}, stderr={}, )
+
+    params = module.params
+
+    state = params['state']
+
+    username = (params['conn_username']).upper()
+    password = params['conn_password']
+    host = params['host']
+    sysnr = params['sysnr']
+    client = params['client']
+
+    task_parameters = params['task_parameters']
+    task_to_execute = params['task_to_execute']
+    task_settings = params['task_settings']
+    tasks_skip = params['tasks_skip']
+
+    if not HAS_ANOTHER_LIBRARY:
+        module.fail_json(
+            msg=missing_required_lib('pyrfc'),
+            exception=ANOTHER_LIBRARY_IMPORT_ERROR)
+
+    # basic RFC connection with pyrfc
+    connection_params = {'user': username, 'passwd': password, 'ashost': host, 'sysnr': sysnr, 'client': client}
+    try:
+        conn = Connection(**connection_params)
+    except Exception as err:
+        result['stderr'] = str(err)
+        result['msg'] = 'Something went wrong connecting to the SAP system.'
+        module.fail_json(**result)
+
+    if task_to_execute is not None:
+        try:
+            raw_params = call_rfc_method(conn, 'STC_TM_SCENARIO_GET_PARAMETERS', {'I_SCENARIO_ID': task_to_execute})
+        except Exception as err:
+            result['stderr'] = str(err)
```

Comment:
I got mixed feelings with the choice of `stderr` here. Ansible associates that with running external commands and capturing `rc`, `stdout` and `stderr`, which is not the case here. In fact, it considers those output keys special and treat them differently, based on that assumption.